### PR TITLE
Fix removing both Relates links

### LINKED ISSUE #487: RemoveIdenticalLink module removes both links
## The Bug
RemoveIdenticalLink module removes both links.

## Example Ticket
https://bugs.mojang.com/browse/MC-199657?focusedCommentId=817403&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-817403

## Expected behavior
Only remove one of the two relates links. (Perhaps always the outwards one?)

### PR DESCRIPTION
**Pull request by SPGoding**
## Purpose
Fix #487.

## Approach
By always removing the outwards link belonging to the smaller ticket key.

#### Checklist
- [x] Included tests
- [ ] Updated documentation in [README](https://github.com/mojira/arisa-kt/blob/master/README.md) and [wiki](https://github.com/mojira/arisa-kt/wiki)
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

File: src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt [hunk 1]
```diff
@@ -3,40 +3,29 @@ package io.github.mojira.arisa.modules
...
-                .values
-                .flatMap { list ->
-                    list.subList(1, list.size).map { it.remove }
+                .filter(::isRelatesLink)
+                .groupBy { it.issue.key }
+                .filterValues { it.size == 2 }
+                .map {
+                    // Always remove the outwards link belonging to the smaller ticket key,
+                    // so that when this module is triggered on both tickets, only one link is removed.
+                    it.value.find { l -> if (key < it.key) l.outwards else !l.outwards }?.remove
```
Comment (chandler05) [inline src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt:21]: I don't think this compares the ticket keys properly, as the key includes the letters and you're trying to compare them.

This doesn't always work properly, for example, "MC-14" is considered less than "MC-2".

I'd recomment comparing after splitting at the dash and converting the leftover string to an int.

Comment (SPGoding) [reply src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt:21]: It really doesn't matter, we just need some consistent way to compare the two keys :eyes:

**Review by urielsalis — DISMISSED** [review]

File: src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt [hunk 2]
```diff
@@ -3,40 +3,29 @@ package io.github.mojira.arisa.modules
...
                 }
 
             assertNotEmpty(removeLinkFunctions).bind()
 
-            removeLinkFunctions.forEach { it.invoke() }
+            removeLinkFunctions.forEach { it?.invoke() }
         }
     }
+
+    private fun isRelatesLink(link: Link): Boolean = link.type.toLowerCase() === "relates"
```
Comment (chandler05) [inline src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt:30]:
```suggestion
    private fun isRelatesLink(link: Link): Boolean = link.type.toLowerCase() == "relates"
```

Comment (SPGoding) [reply src/main/kotlin/io/github/mojira/arisa/modules/RemoveIdenticalLinkModule.kt]: Tysm!

**Review by chandler05 — APPROVED** [review]

**Pull request merged**